src/Underworld.ts

```diff
@@ -719,10 +723,6 @@ export default class Underworld {
         }
         // done processing this unit for this unit's turn
         return false;
-      } else {
-        if (!isUnitsTurn) {
-          u.stamina = 0;
-        }
```

Comment:
This fixed an bug where glops in particular (and a couple others) would desync on multiplayer because they would still have stamina, is this now covered in the above takeAction call?

Reply:
This _should_ be taken care of with takeAction, but I'm not exactly sure what issue/desync you're referring to so I'm not 100% sure.

Let's test for it and if broken, find a solution outside of modifying stamina, since that seems to be causing a lot of issues currently.

Reply:
The problem is, I don't remember why it was introduced so I don't know how to test for it.  What bugs was it causing?

Reply:
I remember now. Previously in gameLoopUnit(), you checked and modified a unit's stamina to decide if it was that unit's turn to move. This also meant setting other units' stamina to 0 in executeNPCTurn() and doing some additional stamina workarounds with smart targeting and stuff like that.

The new subTypesTurn and takeAction thing replaces those stamina checks, making this workaround no longer needed.

Reply:
Okay great! :)